src/libs/pluginprocess/pluginprocess.c

```diff
@@ -291,9 +293,28 @@ int elektraPluginProcessSend (const ElektraPluginProcess * pp, pluginprocess_t c
 	}
 	else
 	{
+		int parentKeyExistsInOriginalKeySet = keySet != NULL ? ksLookup (originalKeySet, key, KDB_O_POP) != NULL : 0;
 		// Copy everything back into the actual keysets
-		keyCopy (key, parentDeserializedKey);
-		if (keySet != NULL) ksCopy (originalKeySet, keySet);
+		// Unfortunately we can't use keyCopy here as ksAppendKey locks it so it will fail
+		// This is the case if the parent key is also contained in the originalKeySet / has been appended
+		// As an invariant we assume plugins don't change the parent key's name during a plugin call
```

Comment:
Yes, that is a perfectly valid assumption but we should also document this in src/plugins/doc.c